circuits/cpp/src/aztec3/circuits/abis/c_bind.cpp

```diff
@@ -376,6 +377,22 @@ WASM_EXPORT void abis__compute_call_stack_item_hash(uint8_t const* call_stack_it
     NT::fr::serialize_to_buffer(get_call_stack_item_hash(call_stack_item), output);
 }
 
+/**
+ * @brief Computes the hash of a message secret for use in l1 -> l2 messaging
+ *
+ * @param secret
+ * @param output
+ */
+WASM_EXPORT void abis__compute_message_secret_hash(uint8_t const* secret, uint8_t* output)
+{
+    NT::fr message_secret;
+    read(secret, message_secret);
+    // TODO(sean): This is not using the generator correctly and is unsafe, update
```

Comment:
wdym? does this have an issue we are tracking it in?

Reply:
Think it is because Noir currently does not support the generators specified as we are doing in cpp. The `message_secret` is to be used in noir, so @Maddiaa0 using this method to have a hash that can be reproduced inside and outside noir.